src/components/tx/security/TransactionSecurityContext/index.tsx

```diff
@@ -0,0 +1,55 @@
+import { type RedefineModuleResponse } from '@/services/security/modules/RedefineModule'
+import { SecuritySeverity } from '@/services/security/modules/types'
+import { type SafeTransaction } from '@safe-global/safe-core-sdk-types'
+import { createContext, useMemo, useState } from 'react'
+import { useRedefine } from '../redefine/useRedefine'
+
+export const TransactionSecurityContext = createContext<{
+  warnings: NonNullable<RedefineModuleResponse['issues']>
+  simulationUuid: string | undefined
+  balanceChange: RedefineModuleResponse['balanceChange']
+  verdict: SecuritySeverity | undefined
+  isLoading: boolean
+  error: Error | undefined
+  needsRiskConfirmation: boolean
+  isRiskConfirmed: boolean
+  setIsRiskConfirmed: (value: boolean) => void
+}>({
+  verdict: SecuritySeverity.NONE,
+  simulationUuid: undefined,
+  warnings: [],
+  balanceChange: undefined,
+  isLoading: false,
+  error: undefined,
+  needsRiskConfirmation: false,
+  isRiskConfirmed: false,
+  setIsRiskConfirmed: () => {},
+})
+
+export const TransactionSecurityProvider = ({
+  children,
+  safeTx,
+}: {
+  children: JSX.Element
+  safeTx: SafeTransaction | undefined
+}) => {
+  const [redefineResponse, redefineError, redefineLoading] = useRedefine(safeTx)
+  const [isRiskConfirmed, setIsRiskConfirmed] = useState(false)
+
+  const providedValue = useMemo(
+    () => ({
+      verdict: redefineResponse?.severity,
```

Comment:
Nit: we use the `SecuritySeverity` enum so I'd name the property the same.

```suggestion
      severity: redefineResponse?.severity,
```